src/Tests/dotnet-new.Tests/TemplateDiscoveryTests.cs

```diff
@@ -0,0 +1,258 @@
+            new DotnetToolCommand(
+                _log,
+                "run",
+                "Microsoft.TemplateSearch.TemplateDiscovery",
+                "--basePath",
+                testDir,
+                "--packagesPath",
+                Path.GetDirectoryName(packageLocation) ?? throw new Exception("Couldn't get package location directory"),
+                "-v")
+                .WithWorkingDirectory(testDir)
+                .Execute()
+                .Should()
+                .ExitWith(0);
+
+            string[] cacheFilePaths = new[]
+            {
+                Path.Combine(testDir, "SearchCache", "NuGetTemplateSearchInfo.json"),
+                Path.Combine(testDir, "SearchCache", "NuGetTemplateSearchInfoVer2.json")
+            };
+            var settingsPath = CreateTemporaryFolder();
+
+            foreach (var cacheFilePath in cacheFilePaths)
+            {
+                Assert.True(File.Exists(cacheFilePath));
+                new DotnetNewCommand(_log)
+                    .WithCustomHive(settingsPath)
+                    .WithEnvironmentVariable("DOTNET_NEW_SEARCH_FILE_OVERRIDE", cacheFilePath)
+                    .WithEnvironmentVariable("DOTNET_CLI_CONTEXT_VERBOSE", "true")
+                    .Execute()
+                    .Should()
+                    .ExitWith(0)
+                    .And.NotHaveStdErr();
+
+                new DotnetNewCommand(_log, "func", "--search")
+                    .WithCustomHive(settingsPath)
+                    .WithEnvironmentVariable("DOTNET_NEW_SEARCH_FILE_OVERRIDE", cacheFilePath)
+                    .WithEnvironmentVariable("DOTNET_CLI_CONTEXT_VERBOSE", "true")
+                    .Execute()
+                    .Should()
+                    .ExitWith(0)
+                    .And.NotHaveStdErr()
+                    .And.NotHaveStdOutContaining("Exception")
+                    .And.HaveStdOutContaining("Microsoft.Azure.WebJobs.ProjectTemplates");
+
+                new DotnetNewCommand(_log)
+                      .WithCustomHive(settingsPath)
+                      .WithEnvironmentVariable("DOTNET_NEW_SEARCH_FILE_OVERRIDE", cacheFilePath)
+                      .WithEnvironmentVariable("DOTNET_CLI_CONTEXT_VERBOSE", "true")
+                      .Execute()
+                      .Should()
+                      .ExitWith(0)
+                      .And.NotHaveStdErr();
+
+                new DotnetNewCommand(_log, "func", "--search")
+                    .WithCustomHive(settingsPath)
+                    .WithEnvironmentVariable("DOTNET_NEW_SEARCH_FILE_OVERRIDE", cacheFilePath)
+                    .WithEnvironmentVariable("DOTNET_CLI_CONTEXT_VERBOSE", "true")
+                    .Execute()
+                    .Should()
+                    .ExitWith(0)
+                    .And.NotHaveStdErr()
+                    .And.NotHaveStdOutContaining("Exception")
+                    .And.HaveStdOutContaining("Microsoft.Azure.WebJobs.ProjectTemplates");
```

Comment:
what is the difference to first 2 processes?

Reply:
It exists in original source. 
It looks like duplicate code. After removing the duplicate one, the test still could succeed.
So I removed the duplicate one.